### PR TITLE
Static site refactor

### PR DESCRIPTION
**Pull request by pellebjerkestrand**
This is quite a large refactoring of the static site generation.

The big win here is a static build that is more similar to the dynamic build (with chunking, minification etc.).

Metadata in the static pages is now proper Yaml frontmatter instead of a custom solution.

The overview of static pages is no longer a custom page, but a standard page with a custom path (`/`).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -119,85 +119,9 @@ These are intended to be used with the [ImageResizer for .NET](https://imageresi
 
 ---
 
-## Project structure
+### ⚙️ Include VS Code tasks?
```
Comment (asbjornh) [inline README.md]: Seeing this again makes me realize how weird this is. We moved this question and logic to `@creuna/cli`, so this package knows nothing about a `tasks.json`. On the other hand, having this documentation in docs for `cli` also seems out of place.

File: static-files/scripts/create-app-components.js [hunk 2]
```diff
@@ -2,13 +2,13 @@
 /* eslint-disable no-console */
 // NOTE: This file is run pre build and creates `app.components.js`.
 const chalk = require('chalk');
-const fs = require('fs');
-const kebabToPascal = require('@creuna/utils/kebab-to-pascal').default;
 const klaw = require('klaw');
 const path = require('path');
-const prettier = require('prettier');
```
Comment (asbjornh) [inline static-files/scripts/create-app-components.js]: Why the removal of prettier?

File: static-files/scripts/create-static-site-files.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+const pagesDirectory = path.join(
+  __dirname,
+  '..',
+  'source',
+  'static-site',
+  'pages'
+);
+
+const disclaimers = [
+  '// NOTE: Do not edit this file. It is built by running `./build/create-static-site-files.js`',
```
Comment (asbjornh) [inline static-files/scripts/create-static-site-files.js]: `./scripts/create-static-site-files.js`

File: templates/package-json/package.json [hunk 4]
```diff
@@ -5,10 +5,10 @@
   "license": "UNLICENSED",
   "scripts": {
     "analyze": "node scripts/create-app-components && webpack --mode=production --env.analyzer",
-    "analyze:static": "node scripts/create-app-components && webpack --progress --mode=production --env.static --env.analyzer",
+    "analyze:static": "node scripts/create-static-site-files && webpack --progress --mode=production --env.static --env.analyzer",
```
Comment (asbjornh) [inline templates/package-json/package.json]: What about `scripts/create-app-components`?

File: static-files/source/static-site/layout.jsx [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const Layout = ({ children }) => children;
+
+Layout.propTypes = {
+  children: PropTypes.node
```
Comment (asbjornh) [inline static-files/source/static-site/layout.jsx]: When a new page is added with `@creuna/react-scripts`, children is `undefined` which generates an annoying warning. Having `null` as the default prop for `children` fixes this

File: static-files/scripts/handle-klaw-error.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+/* eslint-env node */
+/* eslint-disable no-console */
+const chalk = require('chalk');
+
+module.exports = (err, item) => {
+  console.log(
```
[the remark this answers is not in the document]
Comment (pellebjerkestrand) [reply static-files/scripts/handle-klaw-error.js]: It's just refactored (moved), not added.

File: static-files/webpack.config.js [hunk 7]
```diff
@@ -140,55 +135,53 @@ module.exports = (env = {}, options = {}) => {
...
+                  isProduction
+                },
+                paths: require('./source/static-site/pages/paths')
               }),
               new CopyWebpackPlugin(
                 [
-                  { from: 'source/mockup/assets', to: 'mockup/assets' },
-                  { from: 'source/mockup/api', to: 'mockup/api' }
+                  { from: 'source/static-site/assets', to: 'static/assets' },
+                  { from: 'source/static-site/api', to: 'static/api' }
```
[the remark this answers is not in the document]
Comment (pellebjerkestrand) [reply static-files/webpack.config.js]: Sure

File: templates/package-json/package.json (hunk 4 above)
Comment (pellebjerkestrand) [reply templates/package-json/package.json]: Not needed when building a static site. Mounting of the whole page is done in `static-client.js`.

File: static-files/scripts/handle-klaw-error.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (asbjornh) [reply static-files/scripts/handle-klaw-error.js]: Sure, but now that you're refactoring anyway, why not just have an opinion about it?

**Comment by asbjornh** [comment]
> Ref. the `ERROR in TypeError` stuff, @asbjornh: is it a new bug?

Yup! It does not happen in `master`

File: static-files/scripts/handle-klaw-error.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (pellebjerkestrand) [reply static-files/scripts/handle-klaw-error.js]: Because it's out of scope 😛

**Comment by asbjornh** [comment]
Another observation: Removing the import of and reference to `static-site/routes` in `static-server.js` resolves the above issues (with `static-site/routes` still imported in `static-client.js`). This naturally breaks the static site though.

**Comment by asbjornh** [comment]
I tried removing the import of `./pages/pages` in `routes.jsx` and accepting `pages` through props instead, importing `static-site/pages/pages` in `static-client.js` and `static-server.js` and rendering `Routes` with the pages through props. Running WDS and then removing the json import did not throw an error. The `example-page`, however, didn't render anything, and the log of the `pages` object showed this:

```
pages: [ { component: undefined,
    group: 'Eksempelgruppe',
    name: 'Eksempelside',
    path: '/eksempelside' } ]
```

The only time I've ever seen an imported module resolved to `undefined` is when there's a circular reference

**Comment by asbjornh** [comment]
@pellebjerkestrand 
This turns out to be caused by `webpack.optimization.ModuleConcatenationPlugin`. Not completely sure what it does, but removing it resolves the issue.

Seing as restarting WDS also resolved these issues, it's probably safe to use the plugin when building for production

**Comment by asbjornh** [comment]
Can confirm that the `dev` issue no longer exists. Not sure how to mark it as resolved in the GitHub UI

**Comment by pellebjerkestrand** [comment]
> Can confirm that the `dev` issue no longer exists. Not sure how to mark it as resolved in the GitHub UI

I think you need to mark your review as complete.

**Comment by asbjornh** [comment]
> I think you need to mark your review as complete.

Ah. Then I'll wait until something happens with the lodash-thing

**Comment by pellebjerkestrand** [comment]
> > I think you need to mark your review as complete.
> 
> Ah. Then I'll wait until something happens with the lodash-thing

I removed the plugin.

**Comment by asbjornh** [comment]
This fixes #35

**Comment by pellebjerkestrand** [comment]
> > I removed the plugin.
> 
> As far as I can see the babel plugin is still there. Did you forget to push the commit?

Just forgot to remove the dependency 🤦‍♂️

**Comment by asbjornh** [comment]
> Just forgot to remove the dependency 🤦‍♂️

The plugin is still imported in `.babelrc`

**Pull request merged**